File: backend/Models/AiCalls/UsageSummary.cs
namespace Insightboard.Api.Models.AiCalls;

public class UsageSummary
{
    public required int TotalCalls { get; set; }
    public required int InvalidCalls { get; set; }
    public required int TotalInputTokens { get; set; }
    public required int TotalOutputTokens { get; set; }
    public required int WastedTokens { get; set; }
}

File: frontend/src/models/UsageSummary.ts
import z from "zod";

export const UsageSummarySchema = z.object({
  totalCalls: z.number(),
  invalidCalls: z.number(),
  totalInputTokens: z.number(),
  totalOutputTokens: z.number(),
  wastedTokens: z.number(),
});

export type UsageSummary = z.infer<typeof UsageSummarySchema>;
